openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx

```diff
@@ -166,197 +166,85 @@ const GlossaryTermTab = ({ isGlossary, className }: GlossaryTermTabProps) => {
   ]);
   const [confirmCheckboxChecked, setConfirmCheckboxChecked] = useState(false);
 
-  const { paging, handlePagingChange } = usePaging(PAGE_SIZE_LARGE);
-  const [loadingChildren, setLoadingChildren] = useState<
-    Record<string, boolean>
-  >({});
+  const fetchAllTerms = async (loadChildren = false) => {
+    setIsTableLoading(true);
+    try {
+      const key = isGlossary ? 'glossary' : 'parent';
+      // For initial load, only fetch root level terms to improve performance
+      const limit = loadChildren ? API_RES_MAX_SIZE : PAGE_SIZE_MEDIUM;
+      const { data } = await getGlossaryTerms({
+        [key]: activeGlossary?.id || '',
+        limit,
+        fields: [
+          TabSpecificField.OWNERS,
+          TabSpecificField.PARENT,
+          TabSpecificField.CHILDREN,
+        ],
+      });
 
-  const [isLoadingMore, setIsLoadingMore] = useState(false);
-  const [previousGlossaryFQN, setPreviousGlossaryFQN] = useState<
-    string | undefined
-  >(undefined);
-  const [searchTerm, setSearchTerm] = useState('');
-  const [searchInput, setSearchInput] = useState('');
-  const [searchPaging, setSearchPaging] = useState<{
-    offset: number;
-    total?: number;
-    hasMore: boolean;
-  }>({ offset: 0, total: undefined, hasMore: true });
-  const [isExpandingAll, setIsExpandingAll] = useState(false);
-  const [toggleExpandBtn, setToggleExpandBtn] = useState(false);
-
-  const { ref: infiniteScrollRef, inView } = useInView({
-    threshold: 0.1,
-    rootMargin: '50px',
-    trackVisibility: true,
-    delay: 100,
-  });
```

Comment:
@safayavatsal - We shouldn't be removing this. We need infinite scroll in the table. 
So the initial table with first level should follow infinite scroll with page size 50. The above issue is when you expand a child term which has more than 100+ records it is hanging. 

In such cases, when we expand child term, it should load 50 records and then there should be a load more button for expanding next set of 50 records and so on.